test/integration/squash_filter_integration_test.cc

```diff
@@ -0,0 +1,315 @@
+  response->waitForEndStream();
+
+  EXPECT_STREQ("POST", create_stream->headers().Method()->value().c_str());
+  // make sure the env var was replaced
+  ProtobufWkt::Struct actualbody;
+  MessageUtil::loadFromJson(TestUtility::bufferToString(create_stream->body()), actualbody);
+
+  ProtobufWkt::Struct expectedbody;
+  MessageUtil::loadFromJson("{\"spec\": { \"attachment\" : { \"env\": \"" ENV_VAR_VALUE
+                            "\" } , \"match_request\":true} }",
+                            expectedbody);
+
+  EXPECT_TRUE(MessageDifferencer::Equals(expectedbody, actualbody));
+
+  // The second request should be fore the created object
+  EXPECT_STREQ("GET", get_stream->headers().Method()->value().c_str());
+  EXPECT_STREQ("200", response->headers().Status()->value().c_str());
+
+  codec_client_->close();
+  fake_squash_connection->close();
+  fake_squash_connection->waitForDisconnect();
+}
+
+TEST_P(SquashFilterIntegrationTest, ErrorAttaching) {
+
+  IntegrationStreamDecoderPtr response = sendDebugRequest(codec_client_);
+  FakeHttpConnectionPtr fake_squash_connection =
+      fake_upstreams_[1]->waitForHttpConnection(*dispatcher_);
+
+  // respond to create request
+  FakeStreamPtr create_stream =
+      sendSquashCreate(fake_squash_connection, "{\"metadata\":{\"name\":\"oF8iVdiJs5\"},"
+                                               "\"spec\":{\"attachment\":{\"a\":\"b\"},"
+                                               "\"image\":\"debug\",\"node\":\"debug-node\"},"
+                                               "\"status\":{\"state\":\"none\"}}");
+  // respond to read attachment request with error!
+  FakeStreamPtr get_stream = sendSquashOk(
+      fake_squash_connection, "{\"metadata\":{\"name\":\"oF8iVdiJs5\"},\"spec\":{"
+                              "\"attachment\":{\"a\":\"b\"},\"image\":\"debug\",\"node\":"
+                              "\"debug-node\"},\"status\":{\"state\":\"error\"}}");
+
+  response->waitForEndStream();
+
+  EXPECT_STREQ("200", response->headers().Status()->value().c_str());
+
+  codec_client_->close();
+  fake_squash_connection->close();
+  fake_squash_connection->waitForDisconnect();
+}
+
+TEST_P(SquashFilterIntegrationTest, TimeoutAttaching) {
+
+  IntegrationStreamDecoderPtr response = sendDebugRequest(codec_client_);
+  FakeHttpConnectionPtr fake_squash_connection =
+      fake_upstreams_[1]->waitForHttpConnection(*dispatcher_);
+
+  // respond to create request
+  FakeStreamPtr create_stream =
+      sendSquashCreate(fake_squash_connection, "{\"metadata\":{\"name\":\"oF8iVdiJs5\"},"
+                                               "\"spec\":{\"attachment\":{\"a\":\"b\"},"
+                                               "\"image\":\"debug\",\"node\":\"debug-node\"},"
+                                               "\"status\":{\"state\":\"none\"}}");
+  // respond to read attachment. since attachment_timeout is smaller than the squash
+  // attachment_poll_every  config, just one response is enough
```

Comment:
edit _every. I would also finish the sentence explicitely, i.e I would say it is enough because the attachment_timeout will expire before envoy polls the squash server again. Thus the filter will continue iteration